### ISSUE TITLE
Minor code cleanups

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10464]
|<img alt="HelmutG's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| HelmutG reported|
|-|-|
|Trac ID|trac#737|
|Type|enhancement|
|Created|2004-10-05 01:43:22Z|

Attachments:

* [twisted.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/4ca/4cae599309683b04925ec1cb7d79dfab04709af8/d28a367b58eba790fb131d7a60d9534697fcdd09.patch) (8361 bytes) - added by HelmutG on 2004-10-05 01:43:21Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__737 737
type__enhancement enhancement
reporter__HelmutG HelmutG
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1096940602000000 1096940602000000
changetime__1143090974000000 1143090974000000
version__ 
owner__ 
cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
In general, this looks like an improvement.  A couple things, though:

    The patch adds new failure conditions.  This is reasonable, since they seem
to be erroneous cases, but unit tests for these should also be added.

    There are some other behavior changes.  The change from if to while in
ftp.py, for example.  These should have unit tests added as well.

    I don't think changing "del x[0]" to "x.pop(0)" is an improvement.  What's
wrong with "del x[0]"?

    The last hunk in the patch, the part that changes "if df!=-1" to "else:" is
erroneous and should be removed.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to hagbard|
|-|-|

Ready for review in misc-cleanups-737

**Comment by twisted-trac** [comment]
|<img alt="hagbard's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|hagbard commented|
|-|-|

reviewed.

**Comment by twisted-trac** [comment]
|<img alt="hagbard's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|hagbard set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

(In [16409]) Merge misc-cleanups-737

Author: exarkun
Reviewer: hagbard
Fixes [#10464](https://github.com/twisted/twisted/issues/10464)

This applies much of a patch from HelmutG as well as adds several other
random code-cleanups and some new test coverage for the affected code.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
